extract_endpoint/src/crypt_utils.py

```diff
@@ -0,0 +1,13 @@
+import secrets
+import hmac
+import base64
+
+
+def get_salt(num_bytes=64) -> str:
+    return base64.b64encode(secrets.token_bytes(num_bytes)).decode('utf-8')
+
+
+def sign_string(salt: str, key: str, data: str) -> str:
```

Comment:
Is it important that the caller knows to generate a random salt and pass it in, or could this function encapsulate that functionality as well? Maybe making `salt` default to `None` and auto-generate a salt the right way if none is passed in?

Reply:
I'd say it's not that important? ie leave the user the option of not having a salt. The security drops, but not all that much against most types of attack.